### ISSUE TITLE
L027 double raised with wildcard

### ISSUE THREAD

**Issue by jpy-git** [sqlfluff/sqlfluff#1973]
If you lint this query:
```sql
SELECT *
FROM foo
INNER JOIN bar;
```
L027 is raised twice.

N.B. this does not occur if `*` is replaced with `a` so it is specifically the wildcard causing the issue.
 
## Expected Behaviour
L027 only raised once.

## Observed Behaviour
![image](https://user-images.githubusercontent.com/80432516/143318672-d664122f-dd8b-4d37-9d11-158687594bbb.png)

## Steps to Reproduce
lint the query shown above.

## Dialect
Ansi

## Version
Include the output of `sqlfluff --version` along with your Python version
0.8.2 (HEAD of main)

## Configuration
using default config

**Comment by jpy-git** [comment]
Both of these recursive crawls in `get_select_statement_info` are returning the `WildcardIdentifierSegment`:

![image](https://user-images.githubusercontent.com/80432516/143320562-c218b890-dc23-47cc-b372-365dc14241d3.png)

Perhaps the second crawl is unnecessary

**Comment by jpy-git** [comment]
Yep wildcard_identifier inherits from object_reference so I'll remove that second crawl

![image](https://user-images.githubusercontent.com/80432516/143321377-acf0094d-2937-4409-9689-9a742b91eb85.png)

![image](https://user-images.githubusercontent.com/80432516/143321180-77fc4aa4-220a-415d-9823-45301d73cd2b.png)
